pages/slice/language-guide/class-types.md

```diff
@@ -178,6 +178,22 @@ When the generated code decodes a class in sliced format and slices off derived
 slices: it keeps them in the decoded instance. If you later send this class instance to another application that knows
 the derived class type, this application will decode successfully the full type.
 
+## Compact type IDs
+
+When the generated code encodes a class it needs a way to transmit the class's type.
```

Comment:
I feel we should not describe any of these details. The Language Guide is silent on the encoding--it doesn't describe the encoding at all. So why get into the weeds for this aspect of classes?

I'd rather keep it simple with something like

You can make the encoding of a class slightly smaller by assigning a compact type ID to this class. This compact type ID must be a globally unique non-negative integer value. For example:
 ... your example, but with a module ...

This compact type ID affects only the encoding of class instances.

Refer to the Ice Manual (link) to learn more about compact type IDs.

Ice Manual = https://doc.zeroc.com/ice/3.7/the-slice-language/classes/classes-with-compact-type-ids

Reply:
I think this is too simple:
`You can make the encoding of a class slightly smaller by assigning a compact type ID to this class.`
If a user reads this, they're going to come away with one thought: "I should always use compact type IDs"

By explaining it just slightly more, users will actually understand what this is doing, and hence understand it's not a big deal.
But if they're using really long names in really constrained environments, it might be for them.